doc/sphinx/proof-engine/ltac.rst

```diff
@@ -52,127 +66,98 @@ the variable :token:`ident` matches any complex expression with (possible)
 dependencies in the variables :n:`@ident__i` and returns a functional term
 of the form :n:`fun @ident__1 … ident__n => @term`.
 
-The main entry of the grammar is :n:`@ltac_expr`. This language is used in proof
-mode but it can also be used in toplevel definitions as shown below.
+.. note::
+
+   - If you want :n:`@tactic__2; @tactic__3` to be fully applied to the first
+     subgoal generated by :n:`@tactic__1` before applying it to the other
+     subgoals, then you should write:
+
+     - :n:`@tactic__1; [> @tactic__2; @tactic__3 .. ]` rather than
+
+     - :n:`@tactic__1; (@tactic__2; @tactic__3)`.
+
+   .. todo PR I tried to improve the wording for the following bullet,
+      but I'm not quite sure what it refers to.  Perhaps a fool's errand.
+      Does this issue still come up, and if so, exactly which productions?
+
+   - In :token:`tactic_arg_compat`, there is a grammatical ambiguity in
+     how to interpret :token:`qualid`.  It could be either a tactic argument
+     or just a :token:`term`.  This is resolved by checking first whether there
+     is a tactic by that name, then whether it is a reference to a term and
+     using the first one that's found.
+
+     The default interpretation can be overridden by using :n:`ltac:(@qualid)` or
+     :n:`(@qualid)` to force interpretation as, respectively, a tactic or a term.
+
+     .. todo PR With the described behavior, it would never be necessary to use
+        the "ltac:(@qualid)" form since it's checked first.
+
+.. insertprodn ltac_expr tactic_atom
+
+.. prodn::
+   ltac_expr ::= {| @ltac_expr4 | @binder_tactic }
+   let_clause ::= @name := @ltac_expr
+   | @ident {+ @name } := @ltac_expr
+   ltac_expr4 ::= @ltac_expr3 ; {| @ltac_expr3 | @binder_tactic }
+   | @ltac_expr3 ; [ {? @multi_goal_tactics } ]
+   | @ltac_expr3 ; [> {? @multi_goal_tactics } ]
+   | @ltac_expr3
+   multi_goal_tactics ::= {? @ltac_expr } %| @multi_goal_tactics
+   | {? @ltac_expr } .. {? %| {*| {? @ltac_expr } } }
+   | @ltac_expr
+   ltac_expr3 ::= @ltac_expr2
```

Comment:
I'll give explicit lists for these but I think the grammar shouldn't change.  `binder_tactic` appears in several places, so it would have been hard to explain in words, whereas each of the others appears in only one place in `ltac_expr*`.

Reply:
I see the difference but still having two nonterminals for those would help insist on the point that some stuff is not shown in this grammar.